src/api/article/controller.js

```diff
@@ -61,67 +64,173 @@ export const getOne = ({ params: { slug }, user }, res, next) =>
     .then(sendJson(res))
     .catch(next);
 
-export const create = async ({ body }, res, next) => {
-  try {
-    const articleBrandQuery = ArticleBrand.findOne({ slug: body.brand });
-    const articleBrand = (await articleBrandQuery.exec()) || new ArticleBrand({ slug: body.brand });
-    await articleBrand.save();
-
-    const articleCollection = await ArticleCollection.findOne({ slug: body.collectionSlug }).exec();
-
-    const author = await User.findOne({ email: body.authorEmail }).exec();
-
-    const articleBody = {
-      ...body,
-      author: author && author._id,
-      brand: articleBrand._id,
-      collectionId: articleCollection && articleCollection._id,
-    };
-
-    let data;
-    let code;
-    try {
-      const article = Article(articleBody);
-      await article.save();
-      data = serializeArticle(
-        await article
-          .populate('author', POPULATE_OPTIONS.author)
-          .populate('brand', POPULATE_OPTIONS.brand)
-          .populate('collectionId', POPULATE_OPTIONS.collection)
-          .execPopulate()
-      );
-      if (articleCollection) {
-        articleCollection.articles.push(article._id);
-        await articleCollection.save();
-      }
-    } catch (err) {
-      code = 400;
-      data = err;
-    }
-    sendJson(res, code)(data);
-  } catch (err) {
-    next(err);
-  }
-};
+export const create = ({ body }, res, next) =>
```

Comment:
you have a [**promise hell**](https://habrahabr.ru/post/319094/) in this method, should be refactored with `async / await`